src/cpp/session/modules/SessionFiles.cpp

```diff
@@ -660,41 +664,68 @@ Error completeUpload(const core::json::JsonRpcRequest& request,
    
 Error detectZipFileOverwrites(const FilePath& uploadedZipFile,
                               const FilePath& destDir,
-                              json::Array* pOverwritesJson)
+                              const std::string& originalFilename,
+                              json::Array* pOverwritesJson,
+                              bool& rUnzipfound)
 {
-   // query for all of the paths in the zip file
-   core::system::ProcessResult result;
-   Error error = core::system::runCommand("unzip -Z1 " + uploadedZipFile.getAbsolutePath(),
-                                          core::system::ProcessOptions(),
-                                          &result);
-   if (error)
-      return error;
+   // unable to use R's unzip here in worker thread, using system's unzip instead
+   // try a couple locations for unzip, in case it's not on user's PATH
+   std::vector<std::string> pathsToTry({
+                                       "unzip",
+                                       "/usr/bin/unzip"});
```

Comment:
I debated this as well, but went for the multi-line format because it made it more apparent, to me at least, that it was a list of values that could be added to, and not a function call. That's probably overthinking it, but I've fixed the indentation.